src/main/drivers/dshot.c

```diff
@@ -184,92 +187,90 @@ static uint32_t dshot_decode_eRPM_telemetry_value(uint16_t value)
     return (1000000 * 60 / 100 + value / 2) / value;
 }
 
-static void dshot_decode_telemetry_value(uint8_t motorIndex, uint32_t *pDecoded, dshotTelemetryType_t *pType)
+static void dshotDecodeAndSetTelemetryValue(uint32_t *pDecoded, uint16_t *pHighByte, uint8_t motorIndex, uint16_t value, debugType_e debugType)
 {
-    uint16_t value = dshotTelemetryState.motorState[motorIndex].rawValue;
-    const unsigned motorCount = motorDeviceCount();
+    *pDecoded = value & DSHOT_TELEMETRY_VALUE_MASK;
+
+    // Update debug buffer (motorIndex < motorCount guaranteed by caller)
+    if (motorIndex < DEBUG16_VALUE_COUNT) {
+        *pHighByte = dshotTelemetryState.motorState[motorIndex].telemetryData[DSHOT_TELEMETRY_TYPE_STATUS] << SHIFT_BYTE;
+        DEBUG_SET(debugType, motorIndex, *pHighByte | *pDecoded);
+    }
+}
+
+static void dshotDecodeTelemetryValue(uint8_t motorIndex, uint32_t *pDecoded, dshotTelemetryType_t *pType)
+{
+    uint16_t dshotDebugHighByte;
+    const uint16_t value = dshotTelemetryState.motorState[motorIndex].rawValue;
```

Comment:
I can do that, but we are also accessing other properties of the motor state in this functions, should I also pass those as parameters (`telemetryTypes`, `telemetryData`) instead of passing `motorIndex`?

Or do you mean not initalizing the `value` var and passing `dshotTelemetryState.motorState[motorIndex].rawValue` where relevant?

Reply:
Sorry, I thought that all motor-dependent values are for debugging only. telemetryTypes is necessary for processing.